scripts/src/java/org/oppia/android/scripts/maven/GenerateLicenseTexts.kt

```diff
@@ -0,0 +1,435 @@
+    val dependencyList =
+      retrieveDependencyList(mavenDependencyList)
+
+    val dependencyNamesList = retrieveArtifactsNamesList(dependencyList)
+    val dependencyVersionsList = retrieveArtifactsVersionsList(dependencyList)
+
+    writeThirdPartyDependneciesXml(
+      pathToThirdPartyDependenciesXml,
+      dependencyNamesList,
+      dependencyVersionsList,
+      copyrightLicenseSet,
+      dependencyList
+    )
+
+    println("\nScript execution completed successfully.")
+  }
+
+  /**
+   * Retrieve the list of maven dependencies from maven_dependencies.textproto.
+   *
+   * @param pathToPbFile path to the pb file to be parsed
+   *
+   * @return list of [MavenDependency]
+   */
+  private fun retrieveMavenDependencyList(pathToPbFile: String): List<MavenDependency> {
+    return parseTextProto(
+      pathToPbFile,
+      MavenDependencyList.getDefaultInstance()
+    ).mavenDependencyList
+  }
+
+  /**
+   * Helper function to parse the textproto file to a proto class.
+   *
+   * @param pathToPbFile path to the pb file to be parsed
+   * @param proto instance of the proto class
+   *
+   * @return proto class from the parsed textproto file
+   */
+  private fun parseTextProto(
+    pathToPbFile: String,
+    proto: MavenDependencyList
+  ): MavenDependencyList {
+    val protoBinaryFile = File(pathToPbFile)
+    val builder = proto.newBuilderForType()
+    return FileInputStream(protoBinaryFile).use {
+      builder.mergeFrom(it)
+    }.build() as MavenDependencyList
+  }
+
+  private fun retrieveAllLicensesSet(
+    mavenDependencyList: List<MavenDependency>
+  ): Set<CopyrightLicense> {
+    val copyrightLicensesSet = mutableSetOf<CopyrightLicense>()
+    mavenDependencyList.forEach { dependency ->
+      val licenseList = dependency.licenseList
+      if (licenseList.isEmpty()) {
+        throw Exception(MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE)
+      }
+      licenseList.forEach { license ->
+        val licenseText: String
+        val licenseLink: String
+        when (license.verifiedLinkCase) {
+          License.VerifiedLinkCase.SCRAPABLE_LINK -> {
+            licenseText = fetchLicenseText(license.scrapableLink.url)
+            licenseLink = license.scrapableLink.url
+          }
+          License.VerifiedLinkCase.EXTRACTED_COPY_LINK -> {
+            licenseText = fetchLicenseText(license.extractedCopyLink.url)
+            licenseLink = license.extractedCopyLink.url
+          }
+          License.VerifiedLinkCase.DIRECT_LINK_ONLY -> {
+            licenseText = license.directLinkOnly.url
+            licenseLink = license.directLinkOnly.url
+          }
+          else -> throw Exception(MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE)
+        }
+        copyrightLicensesSet.add(
+          CopyrightLicense(
+            license.licenseName,
+            licenseLink,
+            licenseText,
+          )
+        )
+      }
+    }
+    return copyrightLicensesSet
+  }
+
+  private fun retrieveDependencyList(
```

Comment:
This method seems really duplicative to the one above--can we can consolidate some of this common logic into helpers or somehow combine the methods?

Reply:
Made a new helper method for the common part of both methods.